modules/33acrossAnalyticsAdapter.js

```diff
@@ -0,0 +1,486 @@
+
+/**
+ * @typedef {Object} AnalyticsConfig
+ * @property {string} provider - set by pbjs at module registration time
+ * @property {Object} options
+ * @property {string} options.pid - Publisher/Partner ID
+ * @property {string} [options.endpoint=DEFAULT_ENDPOINT] - Endpoint to send analytics data
+ * @property {number} [options.timeout=DEFAULT_TRANSACTION_TIMEOUT] - Timeout for sending analytics data
+ */
+
+/**
+ * @param {AnalyticsConfig} config Analytics module configuration
+ */
+function enableAnalyticsWrapper(config) {
+  const { options } = config;
+
+  const pid = options.pid;
+  if (!pid) {
+    log.error('No partnerId provided for "options.pid". No analytics will be sent.');
+
+    return;
+  }
+
+  const endpoint = calculateEndpoint(options.endpoint);
+  this.getUrl = () => endpoint;
+
+  const timeout = calculateTransactionTimeout(options.timeout);
+  this.getTimeout = () => timeout;
+
+  locals.cache = {
+    pid,
+    auctions: {},
+  };
+
+  const usPrivacy = uspDataHandler.getConsentData();
+  if (/^1[Y|N|-]{3}$/.test(usPrivacy)) {
+    locals.cache.usPrivacy = usPrivacy;
+  }
+
+  analyticsAdapter.originEnableAnalytics(config);
+}
+
+/**
+ * @param {string} [endpoint]
+ * @returns {string}
+ */
+function calculateEndpoint(endpoint = DEFAULT_ENDPOINT) {
+  if (typeof endpoint === 'string' && endpoint.startsWith('http')) {
+    return endpoint;
+  }
+
+  log.info(`Invalid endpoint provided for "options.endpoint". Using default endpoint.`);
+
+  return DEFAULT_ENDPOINT;
+}
+/**
+ * @param {number} [configTimeout]
+ * @returns {number} Transaction Timeout
+ */
+function calculateTransactionTimeout(configTimeout = DEFAULT_TRANSACTION_TIMEOUT) {
+  if (typeof configTimeout === 'number' && configTimeout >= 0) {
+    return configTimeout;
+  }
+
+  log.info(`Invalid timeout provided for "options.timeout". Using default timeout of ${DEFAULT_TRANSACTION_TIMEOUT}ms.`);
+
+  return DEFAULT_TRANSACTION_TIMEOUT;
+}
+
+/** necessary for testing */
+analyticsAdapter.originDisableAnalytics = analyticsAdapter.disableAnalytics;
+analyticsAdapter.disableAnalytics = function () {
+  analyticsAdapter._oldEnable = enableAnalyticsWrapper;
+  locals.reset();
+  analyticsAdapter.originDisableAnalytics();
+};
+
+adapterManager.registerAnalyticsAdapter({
+  adapter: analyticsAdapter,
+  code: PROVIDER_NAME,
+  gvlid: GVLID,
+});
+
+export default analyticsAdapter;
+
+/**
+ * @param {AnalyticsCache} analyticsCache
+ * @param {string} completedAuctionId value of auctionId
+ * @return {AnalyticsReport} Analytics report
+ */
+function createReportFromCache(analyticsCache, completedAuctionId) {
+  const { pid, auctions, usPrivacy } = analyticsCache;
+
+  const report = {
+    pid,
+    src: 'pbjs',
+    analyticsVersion: ANALYTICS_VERSION,
+    pbjsVersion: '$prebid.version$', // Replaced by build script
+    auctions: [ auctions[completedAuctionId] ],
+  }
+  if (usPrivacy) {
+    report.usPrivacy = usPrivacy;
+  }
```

Comment:
could you also add support for GPP please.. but don't make those changes part of this PR.. have them in a separate branch.

https://github.com/33Across/Prebid.js/pull/64 shows how we added GPP support in our UIM module

Reply:
I *almost* did that, but then thought I better wait. Yes, I'll add that, as well as COPPA support, in two new branches